File: ExerciceAnimation/app/src/main/java/com/example/animation/MainActivity.java
package com.example.animation;

import android.animation.AnimatorSet;
import android.animation.ArgbEvaluator;
import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.graphics.Color;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    AnimatorSet animatorSet;
    View view;
    Button start;
    boolean isAnimating = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        view = findViewById(R.id.view);
        start = findViewById(R.id.button);

        ObjectAnimator rotation = ObjectAnimator.ofFloat(view, View.ROTATION, 0f, 360f);
        rotation.setDuration(2000);
        rotation.setRepeatCount(ObjectAnimator.INFINITE);

        ObjectAnimator colorChange = ObjectAnimator.ofArgb(view, "backgroundColor", Color.RED, Color.BLUE);
        colorChange.setDuration(2000);
        colorChange.setEvaluator(new ArgbEvaluator());
        colorChange.setRepeatCount(ObjectAnimator.INFINITE);
        colorChange.setRepeatMode(ObjectAnimator.REVERSE);

        ObjectAnimator translation = ObjectAnimator.ofFloat(view, View.TRANSLATION_X, 0f, 500f);
        translation.setDuration(2000);
        translation.setRepeatCount(ObjectAnimator.INFINITE);
        translation.setRepeatMode(ObjectAnimator.REVERSE);

        animatorSet = new AnimatorSet();
        animatorSet.playTogether(rotation, colorChange, translation);

        start.setOnClickListener(source -> {
            if (isAnimating) {
                animatorSet.pause();
                isAnimating = false;
            } else {
                animatorSet.start();
                isAnimating = true;
            }
        });
    }
}

File: projetDentiste/app/src/main/java/com/factures/applicationpaiementfactures/MainActivity.kt
package com.factures.applicationpaiementfactures

import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import android.widget.EditText

import android.widget.LinearLayout

import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat


class MainActivity : AppCompatActivity() {

    lateinit var dent1: LinearLayout
    lateinit var dent2: LinearLayout
    lateinit var numDent1: EditText
    lateinit var numDent2: EditText
    lateinit var noteDent1: EditText
    lateinit var noteDent2: EditText
    lateinit var canalDent1: CheckBox
    lateinit var canalDent2: CheckBox


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val singleton = Singleton.getInstance(applicationContext)

        dent1 = findViewById(R.id.dent1)
        dent2 = findViewById(R.id.dent2)
        numDent1 = findViewById(R.id.numDent1)
        numDent2 = findViewById(R.id.numDent2)
        noteDent1 = findViewById(R.id.noteDent1)
        noteDent2 = findViewById(R.id.noteDent2)
        canalDent1 = findViewById(R.id.canalDent1)
        canalDent2 = findViewById(R.id.canalDent2)


        try {
            val temp = singleton.deserialiserListe()
            singleton.listeDent = temp
            noteDent1.setText(singleton.listeDent[0].memo)
            numDent1.setText(singleton.listeDent[0].numDent)
            canalDent1.isChecked = singleton.listeDent[0].check

            noteDent2.setText(singleton.listeDent[1].memo)
            numDent2.setText(singleton.listeDent[1].numDent)
            canalDent2.isChecked = singleton.listeDent[1].check
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onStop() {
        super.onStop()

        val memo = noteDent1.text.toString()
        val numDent = numDent1.text.toString()
        val canalDent = canalDent1.isChecked()
        Singleton.getInstance(applicationContext).ajouterDent(Dent(memo, numDent, canalDent))

        val memo2 = noteDent2.text.toString()
        val numDent2 = numDent2.text.toString()
        val canalDent2 = canalDent2.isChecked()
        Singleton.getInstance(applicationContext).ajouterDent(Dent(memo2, numDent2, canalDent2))

        try {
            Singleton.getInstance(applicationContext).serialiserListe()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}